tests/unit/core/reducers/test_abuse.js

```diff
@@ -0,0 +1,164 @@
+        reporter: null,
+      })));
+
+      expect(store.getState().abuse).toMatchObject({
+        bySlug: {
+          'another-addon': {
+            message: 'The add-on is boring',
+          },
+          'some-addon': {
+            message: 'This add-on is malwaré.',
+          },
+        },
+      });
+    });
+
+    it('throws an error if multiple reports are submitted for the same add-on', () => {
+      const { store } = dispatchClientMetadata();
+
+      store.dispatch(loadAddonAbuseReport(createFakeAddonAbuseReport({
+        addon: { ...fakeAddon, slug: 'some-addon' },
+        message: 'This add-on is malwaré.',
+        reporter: null,
+      })));
+
+      expect(() => {
+        store.dispatch(sendAddonAbuseReport({
+          addonSlug: 'some-addon',
+          errorHandlerId: 'test-abuse',
+          message: 'The add-on is boring',
+        }));
+      }).toThrow('You already reported this add-on');
+    });
+
+    describe('sendAddonAbuseReport', () => {
+      let defaultParams;
+
+      beforeEach(() => {
+        defaultParams = {
+          addonSlug: 'some-addon',
+          errorHandlerId: 'some-error-handler',
+          message: 'The add-on is malware',
+        };
+      });
+
+      it('dispatches a sendAddonAbuseReport', () => {
+        const action = sendAddonAbuseReport(defaultParams);
+
+        expect(action.type).toEqual(SEND_ADDON_ABUSE_REPORT);
+        expect(action.payload).toEqual(defaultParams);
+      });
+
+      it('requires an addonSlug', () => {
+        expect(() => {
+          const partialParams = { ...defaultParams };
+          delete partialParams.addonSlug;
+
+          sendAddonAbuseReport(partialParams);
+        }).toThrow('addonSlug is required');
+      });
+
+      it('requires an errorHandlerId', () => {
+        expect(() => {
+          const partialParams = { ...defaultParams };
+          delete partialParams.errorHandlerId;
+
+          sendAddonAbuseReport(partialParams);
+        }).toThrow('errorHandlerId is required');
+      });
+
+      it('requires a message', () => {
+        expect(() => {
+          const partialParams = { ...defaultParams };
+          delete partialParams.message;
+
+          sendAddonAbuseReport(partialParams);
+        }).toThrow('message is required');
+      });
+    });
+
+    describe('loadAddonAbuseReport', () => {
+      let addon;
+      let message;
+      let response;
+
+      beforeEach(() => {
+        addon = { ...fakeAddon, slug: 'Ego' };
+        message = 'I am Groot!';
+        response = createFakeAddonAbuseReport({ addon, message });
+      });
+
+      it('saves the abuse report response to the reducer', () => {
+        const state = abuseReducer(
+          initialState, loadAddonAbuseReport(response));
+
+        expect(state.bySlug[addon.slug].message).toEqual(message);
+      });
+
+      it('requires an addon', () => {
+        expect(() => {
+          loadAddonAbuseReport({
```

Comment:
These tests will be easier to maintain if they use the same approach as above:

```js
const partialParams = { ...defaultParams };
delete partialParams.addon;
loadAddonAbuseReport(partialParams);
```

Reply:
True, I think I just got lazy. Will do!